test/integration/local/test_mnist_training.py

```diff
@@ -28,40 +27,20 @@
 TEST_INPUT = 'file://{}'.format(os.path.join(MNIST_PATH, 'test'))
 
 
-def test_mnist_training_and_serving(docker_image, sagemaker_local_session, local_instance_type,
-                                    framework_version, tmpdir):
+def test_single_machine(docker_image, sagemaker_local_session, local_instance_type,
+                        framework_version, tmpdir):
     mx = MXNet(entry_point=SCRIPT_PATH, role='SageMakerRole', train_instance_count=1,
                train_instance_type=local_instance_type, sagemaker_session=sagemaker_local_session,
                image_name=docker_image, framework_version=framework_version,
                output_path='file://{}'.format(tmpdir))
 
     _train_and_assert_success(mx, str(tmpdir))
 
-    with local_mode_utils.lock():
-        try:
-            model = mx.create_model(model_server_workers=NUM_MODEL_SERVER_WORKERS)
-            predictor = _csv_predictor(model, local_instance_type)
-            data = numpy.zeros(shape=(1, 1, 28, 28))
-            prediction = predictor.predict(data)
-        finally:
-            mx.delete_endpoint()
 
-    # Check that there is a probability for each possible class in the prediction
-    prediction_values = prediction.decode('utf-8').split(',')
-    assert len(prediction_values) == 10
+def test_distributed(docker_image, sagemaker_local_session, framework_version, processor, tmpdir):
+    if processor == 'gpu':
+        pytest.skip('Local Mode does not support distributed training on GPU.')
```

Comment:
does this fail when using gpu?

Reply:
yeah. see previous builds on this PR (also ran into this last week when doing local runs)